lib/services/android-project-service.ts

```diff
@@ -265,14 +248,64 @@ class AndroidProjectService implements IPlatformProjectService {
 
 			let libProjProp = path.join(libraryPath, "project.properties");
 			if (this.$fs.exists(libProjProp).wait()) {
-				this.updateProjectReferences(platformData.projectRoot, targetLibPath);
+				this.updateProjectReferences(platformData.projectRoot, targetLibPath).wait();
 			}
 		}).future<void>()();
 	}
-
+	
 	public getFrameworkFilesExtensions(): string[] {
 		return [".jar", ".dat"];
 	}
+	
+	public preparePluginNativeCode(pluginData: IPluginData): IFuture<void> {
+		return (() => {
+			let pluginPlatformsFolderPath = this.getPluginPlatformsFolderPath(pluginData);
+			
+			// Handle *.jars inside libs folder
+			let libsFolderPath = path.join(pluginPlatformsFolderPath, AndroidProjectService.LIBS_FOLDER_NAME);
+			if(this.$fs.exists(libsFolderPath).wait()) {
+				let libsFolderContents = this.$fs.readDirectory(libsFolderPath).wait();
+				_(libsFolderContents)
+					.filter(libsFolderItem => path.extname(libsFolderItem) === ".jar")
+					.map(jar => this.addLibrary(this.platformData, path.join(libsFolderPath, jar)).wait());
```

Comment:
You are using lodash's chain, so you should add `.value()` at the end.
Or you can use
```TypeScript
libsFolderContents.filter(libsFolderItem => path.extname(libsFolderItem) === ".jar")
              .map(jar => this.addLibrary(this.platformData, path.join(libsFolderPath, jar)).wait());
```